google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/QueryParameterValue.java

```diff
@@ -330,32 +359,40 @@ public static QueryParameterValue dateTime(String value) {
       case ARRAY:
         throw new IllegalArgumentException("Cannot convert ARRAY to String value");
       case TIMESTAMP:
-        if (value instanceof Long) {
-          return timestampFormatter.print(((Long) value) / 1000);
+        if (value instanceof Timestamp) {
+          return ((Timestamp) value).formatMicroCloudTime();
+        } else if (value instanceof Long) {
+          return Timestamp.ofTimeMicroseconds((Long) value).formatMicroCloudTime();
         } else if (value instanceof String) {
           // verify that the String is in the right format
-          timestampFormatter.parseMillis((String) value);
```

Comment:
Can we remove the xxxFormatter now? The functionality seems overlapping with the new date classes.